mmedit/models/editors/inst_colorization/colorization_net.py

```diff
@@ -0,0 +1,313 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import torch
+import torch.nn as nn
+from mmengine.model import BaseModule
+
+from mmedit.registry import MODULES
+from .weight_layer import get_norm_layer
+
+
+@MODULES.register_module()
+class ColorizationNet(BaseModule):
+    """Real-Time User-Guided Image Colorization with Learned Deep Priors. The
+    backbone used for.
+
+    https://arxiv.org/abs/1705.02999
+
+    Codes adapted from 'https://github.com/ericsujw/InstColorization.git'
+    'InstColorization/blob/master/models/networks.py#L108'
+
+    Args:
+        input_nc (int): input image channels
+        output_nc (int): output image channels
+        norm_type (str): instance normalization or batch normalization
+        use_tanh (bool): Whether to use nn.Tanh() Default: True.
+        classification (bool): backprop trunk using classification,
+            otherwise use regression. Default: True
+    """
+
+    def __init__(self,
+                 input_nc,
+                 output_nc,
+                 norm_type,
+                 use_tanh=True,
+                 classification=True):
```

Comment:
Type lint in the function signature is missing. (Maybe can convert to a **good first issue**)

Reply:
Agree